frame/dapps-staking/src/fix.rs

```diff
@@ -0,0 +1,175 @@
+                        .or_default();
+
+                    // update total and number of stakers
+                    contract_stake_info.total =
+                        contract_stake_info.total.saturating_add(staked_value);
+                    contract_stake_info.number_of_stakers += 1;
+                    // check for total consumed weight
+                    // store last processed contract and break out if the limit is reached
+                    if consumed_weight >= weight_limit {
+                        log::info!(
+                            ">>> Ledger migration stopped after consuming {:?} weight.",
+                            consumed_weight
+                        );
+                        restake_fix.last_processed_staker_contract =
+                            Some(contract_address.clone().to_vec());
+                        break 'staker_iter;
+                    }
+                }
+
+                // when one staker is processed, store it and clear stored contract
+                restake_fix.last_processed_staker_contract = None;
+                restake_fix.last_fully_processed_staker =
+                    Some(Ledger::<T>::storage_map_final_key(staker));
+            }
+        } else {
+            log::info!(">>> Starting to empty the accumulator");
+            // if contractStakeInfo is empty, we're done
+            if restake_fix.contract_staking_info.is_empty() {
+                log::info!(">>> Accumulator empty, all done.");
+            } else {
+                // get current era and weight for that read
+                let current_era = Pallet::<T>::current_era();
+                consumed_weight = consumed_weight.saturating_add(T::DbWeight::get().reads(1));
+
+                let staking_info_iter = restake_fix.contract_staking_info.iter();
+
+                for (contract, contract_stake_info) in staking_info_iter {
+                    // write new ContractEraStake from corrected records in contract_staking_info
+                    // and remove that record afterwards
+                    // break out when the weight hits the limit
+                    ContractEraStake::<T>::insert(&contract, current_era, contract_stake_info);
+                    restake_fix.contract_staking_info.remove(contract);
+                    consumed_weight = consumed_weight.saturating_add(T::DbWeight::get().writes(1));
+                    if consumed_weight >= weight_limit {
+                        log::info!(
+                            ">>> Contract info migration stopped after consuming {:?} weight.",
+                            consumed_weight
+                        );
+                        break;
+                    }
+                }
+            }
+        }
+        // store updated accumulator and update total weight for that write
+        RestakeFixAccumulator::<T>::put(restake_fix);
+        consumed_weight = consumed_weight.saturating_add(T::DbWeight::get().writes(1));
```

Comment:
Imo, this is incorrect.

If you've reached this point, you should kill `RestakeFixAccumulator` since it should mean you're finished.

You should also disable maintenance mode.